pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +49,51 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
+// cgroup.
+func cgrpSanityCheck() {
+
+	procCount := 0
+	err := filepath.Walk(cgroupRoot, func(path string, info fs.FileInfo, err error) error {
```

Comment:
This function returns `nil` when it fails to traverse the cgroup tree _or_ when it finds more than 2 processes attached. I understand the objective, but would not it be more readable to have it return a boolean? Something like
```go
        lowProcCount := filepath.Walk(cgrouRoot, func( ... ) bool {
```

Reply:
Makes sense, but the function type is defined by the go library: https://golang.org/pkg/path/filepath/#Walk

The reason is that they have a special error that you can return on a directory to skip it in the traversing.

Reply:
Ah right, apologies. Let's keep the error then.